### PR TITLE
use directory-mapping instead of simple containerd.socket-file-mapping

### LINKED ISSUE #632: falco will lose connection to containerd.socket if containerd on host gets restarted
**Describe the bug**

If the containerd-process on the host-system gets restarted. Falco will loose the connection to the socket.

**How to reproduce it**

* Increase log-level of falco to show libs-debug
* Restart containerd on the host
* Logs will show failing containerd-socket tries
* falco will continue to try to reconnect, but it will never work

**Expected behaviour**

Falco has no issue when containerd gets restarted on host

**Additional context**

The issue is caused by mapping the socket as a simple file into the pod. These mappings are done inode-based, so a restarting containerd-process will drop&create a new socket-file on the host, causing a new inode to be used. The inode within the pod will no longer point to the socket-file.

Fix: Instead map the socket-file-containing-folder into the pod, so falco detects the broken socket, retries and is able to use the new inode of the socket.

FYI: A suitable patch to provide this functionality is on the way.

### PR DESCRIPTION
**Pull request by devfaz**
**What type of PR is this?**

/kind bug

/kind chart-release

**Any specific area of the project related to this PR?**

/area falco-chart

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #632 

**Special notes for your reviewer**:

to allow falco to reconnect if containerd got restarted on host

**Checklist**

- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] CHANGELOG.md updated

### CODE REVIEW

**Comment by alacuku** [comment]
Hi @devfaz, thank you for the PR!

Could you please update the docs by running `make docs`?

**Comment by devfaz** [comment]
> Hi @devfaz, thank you for the PR!
> 
> Could you please update the docs by running `make docs`?

run `make docs` but it didnt generate any change.

**Comment by alacuku** [comment]
> > Hi @devfaz, thank you for the PR!
> > Could you please update the docs by running `make docs`?
> 
> run `make docs` but it didnt generate any change.

Running `make docs` on your branch:

```bash
diff --git a/charts/falco/README.md b/charts/falco/README.md
index e83c3f5..81ecd66 100644
--- a/charts/falco/README.md
+++ b/charts/falco/README.md
@@ -581,7 +581,7 @@ If you use a Proxy in your cluster, the requests between `Falco` and `Falcosidek
 
 ## Configuration
 
-The following table lists the main configurable parameters of the falco chart v4.2.2 and their default values. See [values.yaml](./values.yaml) for full list.
+The following table lists the main configurable parameters of the falco chart v4.2.3 and their default values. See [values.yaml](./values.yaml) for full list.

```

File: charts/falco/templates/pod-template.tpl [hunk 1]
```diff
@@ -65,7 +65,7 @@ spec:
         {{- if .enabled }}
         {{- if .containerd.enabled }}
         - --cri
-        - /run/containerd/containerd.sock
+        - /run/containerd/{{ base .containerd.socket }}
```
Comment (alacuku) [inline charts/falco/templates/pod-template.tpl:68]: Could you do the same for `docker` and `crio` container runtimes?

Comment (devfaz) [reply charts/falco/templates/pod-template.tpl:72]: Done - sorry for the delay :|

**Comment by alacuku** [comment]
@devfaz, do you plan to update this PR by any chance?

**Comment by devfaz** [comment]
@alacuku I already did? Do you miss anything?

**Comment by alacuku** [comment]
You need to rebase and resolve the conflicts.

**Comment by devfaz** [comment]
> You need to rebase and resolve the conflicts.

Done, sorry havnt seen it.

**Review by alacuku — APPROVED** [review]
/lgtm

**Pull request merged**